aws/kms.go

```diff
@@ -0,0 +1,54 @@
+package aws
+
+import (
+	b64 "github.com/hairyhenderson/gomplate/base64"
+
+	"github.com/aws/aws-sdk-go/service/kms"
+)
+
+// KMS -
+type KMS struct {
+	Client *kms.KMS
+}
+
+// NewKMS - Create new KMS client
+func NewKMS(option ClientOptions) *KMS {
+	client := kms.New(SDKSession())
+	return &KMS{
+		Client: client,
+	}
+}
+
+// Encrypt plaintext using the specified key.
+// Returns a base64 encoded ciphertext
+func (k *KMS) Encrypt(keyID, plaintext string) (string, error) {
+	input := &kms.EncryptInput{
+		KeyId:     &keyID,
+		Plaintext: []byte(plaintext),
+	}
+	output, err := k.Client.Encrypt(input)
+	if err != nil {
+		return "", err
+	}
+	ciphertext, err := b64.Encode(output.CiphertextBlob)
+	if err != nil {
```

Comment:
@hairyhenderson Because the `base64.Encode` function can potentially return an error, I had to handle it here. I am however wondering if this is necessary at all, is there a case when the gomplate `base64.Encode` would actually return an error?

https://github.com/hairyhenderson/gomplate/blob/2bab08cea50a3130d2e6fea85589bf2d1bf173d1/base64/base64.go#L8-L10

Reply:
🤔 😂 clearly not... Looks like that may have gotten caught up in the mega-PR that was #380. It may have made sense to me at the time as a hedge against some future change. This _may_ fulfil some interface too, though I can't think which. And maybe it was just to "balance" out the `Decode` function, which can and does throw `error`s.

_However,_ removing the `error` from `Encode` will break API, so I'm inclined to leave it here. And so, not knowing whether there ever _will_ be an `error`, I think your error-handling code should remain 😉.